api/proto/config/devmodel.proto

```diff
@@ -171,3 +156,152 @@ message PhysicalIO {
+  // Creates aggregation groups that share the same speed and duplex settings.
+  // Utilizes all slaves in the active aggregator according to the 802.3ad
+  // specification.
+  // Requires that upstream networks equipment supports and has the IEEE 802.3ad
+  // Dynamic link aggregation properly configured.
+  BOND_MODE_802_3AD = 5;
+  // BOND_MODE_BALANCE_TLB : adaptive transmit load balancing. Channel bonding that
+  // does not require any special switch support. The outgoing traffic
+  // is distributed according to the current load (computed relative to the speed)
+  // on each slave.
+  // Incoming traffic is received by the current slave. If the receiving slave
+  // fails, another slave takes over the MAC address of the failed receiving slave.
+  // This mode provides (TX) load balancing and fault tolerance.
+  BOND_MODE_BALANCE_TLB = 6;
+  // BOND_MODE_BALANCE_ALB : adaptive load balancing. Includes BOND_MODE_BALANCE_TLB
+  // plus receive load balancing (RLB) for IPV4 traffic, and does not require
+  // any special switch support. The receive load balancing is achieved by ARP
+  // negotiation.
+  // This mode provides load balancing and fault tolerance.
+  BOND_MODE_BALANCE_ALB = 7;
+}
+
+// Option specifying the rate in which EVE will ask LACP link partners
+// to transmit LACPDU packets in 802.3ad mode.
+enum LacpRate {
+  // Default LACP rate is Slow.
+  LACP_RATE_UNSPECIFIED = 0;
+  // Request LACP partner to transmit LACPDUs every 30 seconds.
+  LACP_RATE_SLOW = 1;
+  // Request LACP partner to transmit LACPDUs every 1 second.
+  LACP_RATE_FAST = 2;
+}
+
+// BondAdapter aggregates multiple PhysicalIOs into one adapter for redundancy
+// and load-spreading purposes.
+message BondAdapter {
+  // Name of this bond adapter.
+  string logicallabel = 1;
+
+  // A physical name of the bond interface.
+  // Note that the interface name is limited in Linux kernel to 15 characters.
+  // If not defined, logicallabel will be used instead.
+  string interface_name = 2;
+
+  // Logical names of aggregated PhysicalIOs.
+  repeated string lower_layer_names = 3;
+
+  // A bonding mode specifies the policy indicating how bonding slaves are used
+  // during network transmission
+  BondMode bond_mode = 4;
+
+  // Method by which link failures and recoveries are detected.
+  oneof monitoring {
+    // The MII monitor is driver-dependent. It monitors the links from the device
+    // to the nearest connected switch. If the failure occurs beyond the nearest
+    // connected switch, it cannot be detected by MII monitor.
+    MIIMonitor mii = 5;
+    // The ARP monitor is based on the communication to the target
+    // hosts designated by their IP addresses. Even if the link is beyond
+    // the nearest connected switch, the APR monitor can detect it.
+    ArpMonitor arp = 6;
+  }
+
+  // Logical name of one of the aggregated PhysicalIO to prefer
+  // in the Active-Backup mode.
+  string primary_physical_io = 7;
```

Comment:
What is the relationship between this and the lower_layer_names? Does the user have to specify a primary?

Also, calling it "lower_layer" in one place and "physical_io" here makes it harder to tell that this should be one of the lower_layer_names above.
Does the API have to specify one or will EVE pick one if not set?

Reply:
I have decided to simplify this and remove `primary_physical_io`. Now EVE will pick the first adapter from the `lower_layer_names` list as the primary one (I've updated the comment for this field to mention that the list order is relevant for Active-Backup mode, but otherwise it isn't).